pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -127,3 +129,88 @@ def close(self) -> None:
         if not self.skip_check_coredump:
             self._check_coredump()
         super().close()
+
+    Case = namedtuple('Case', 'index name type keyword group attribute subcase')
+
+    def get_test_menu(
+        self, regex: str = "Here's the test menu, pick your combo:(.+)Enter test for running.", trigger: str = ''
+    ) -> List[Case]:
+        """
+        Get test case list from test menu via UART print.
+
+        This function should be called after `Press ENTER to see the list of tests` prompt is shown.
+
+        Args:
+            regex: Regular expression to match the output from chip, menu block should be in the first group.
+            trigger: Keys to trigger chip to print test menu by UART.
+
+        Returns:
+            A `list` of avaliable test cases, in `Case`. `Case` has following fields:
+                - index: The index of the case, which can be used to run this case.
+                - name: The name of the case.
+                - type: Type of this case, which can be `normal` `multi_stage` or `multi_device`.
+                - keyword: List of additonal keywords of this case. For now, we have `disable` and `ignored`.
+                - group: List of groups of this case, this is usually the component which this case belongs to.
+                - attribute: List of attributes of this case, which is used to describe timeout duration,
+                    test environment, etc.
+                - subcase: List of subcases of this case, if this case is a `multi_stage` or `multi_device` one.
+        """
+        self.write(trigger)
+        menu_block = self.expect(regex).group(1)
+        cases = str(menu_block, encoding='UTF-8').splitlines()
+
+        case_regex = re.compile(r'\((\d+)\)\s\"(.+)\"\s(\[.+\])+')
+        subcase_regex = re.compile(r'\t\((\d+)\)\s\"(.+)\"')
+
+        test_menu = []
+        for case in cases:
+            case_match = case_regex.match(case)
+            if case_match is not None:
```

Comment:
here raise an exception when "if case_match is None and subcase_match is None" would make It easier to find those test menus with unexpected output.